python/lsst/resources/http.py

```diff
@@ -1137,6 +1218,116 @@ def walk(
+
+        Returns
+        -------
+        url : `str`
+            HTTP URL signed for PUT.
+        """
+        return self._sign_with_macaroon(ActivityCaveat.UPLOAD, expiration_time_seconds)
+
+    def to_fsspec(self) -> tuple[AbstractFileSystem, str]:
+        """Return an abstract file system and path that can be used by fsspec.
+
+        Returns
+        -------
+        fs : `fsspec.spec.AbstractFileSystem`
+            A file system object suitable for use with the returned path.
+        path : `str`
+            A path that can be opened by the file system object.
+        """
+        if self.isdir():
+            raise NotImplementedError(
+                f"method HttpResourcePath.to_fsspec() not implemented for directory {self}"
+            )
+
+        if fsspec is None:
+            raise ImportError("fsspec is not available")
+
+        async def get_client(**kwargs: Any) -> aiohttp.ClientSession:
+            # Create a SSL context equipped with the required certificates of
+            # the trusted certificate authorities. fsspect client must use
+            # that context to download the file.
+            ssl_context = ssl.create_default_context()
+            if self._config.ca_bundle is not None:
+                ssl_context.load_verify_locations(capath=self._config.ca_bundle)
+            return aiohttp.ClientSession(connector=aiohttp.TCPConnector(ssl=ssl_context), **kwargs)
+
+        # Retrieve a signed URL for a duration of 1 hour. We assume the URL
+        # is to be used only for downloading, to avoid the risk the caller
+        # modify its contents inadvertently.
+        url = self.generate_presigned_get_url(expiration_time_seconds=3_600)
+
+        # We are returning a a fsspec built-in filesystem implemented by
+        # class 'fsspec.implementations.http.HTTPFileSystem'. The constructor
+        # of this class accepts the argument 'block_size'. The default
+        # value is 'fsspec.utils.DEFAULT_BLOCK_SIZE' which is 5 MB.
+        # That seems to be a reasonable block size for downloading files.
+        return fsspec.core.url_to_fs(url, get_client=get_client)
+
+    def _sign_with_macaroon(self, activity: ActivityCaveat, expiration_time_seconds: int) -> str:
+        # dCache and XrootD servers support delivering of macaroons
+        if self.server is None:
+            raise NotImplementedError(f"server for '{self}' does not support signing URLs")
+
+        # dCache and XrootD servers are known to support delivery of macaroons.
+        if self.server not in ("dcache", "xrootd"):
+            raise NotImplementedError(f"server '{self._server}' does not support signing for {self}")
+
+        match activity:
+            case ActivityCaveat.DOWNLOAD:
+                activity_caveat = "DOWNLOAD,LIST"
+            case ActivityCaveat.UPLOAD:
+                activity_caveat = "UPLOAD,LIST"
+
+        # Request a macaroon for the requested activities and duration duration
```

Comment:
It might be helpful to include a link to some documentation explaining this macaroon stuff...

I eventually found https://www.dcache.org/manuals/workshop-2017-05-29-Umea/000-Final/anupam_macaroons_v02.pdf which helped me figure out what was going on but it took a minute.

Reply:
✅ Done: pointer to official documentation added.